src/mongo_logger.ts

```diff
@@ -143,37 +180,234 @@ function resolveLogPath(
   return process.stderr;
 }
 
+export interface Log extends Record<string, any> {
+  s: SeverityLevel;
+  t: Date;
+  c: MongoLoggableComponent;
+  message?: string;
+}
+
+export interface MongoDBLogWritable {
+  write(log: Log): unknown;
+}
+
+function compareSeverity(s0: SeverityLevel, s1: SeverityLevel): 1 | 0 | -1 {
+  const s0Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s0);
+  const s1Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s1);
+
+  return s0Num < s1Num ? -1 : s0Num > s1Num ? 1 : 0;
+}
+
+function DEFAULT_LOG_TRANSFORM(logObject: Record<string, any>): Omit<Log, 's' | 't' | 'c'> {
+  let log: Omit<Log, 's' | 't' | 'c'> = {};
+
+  const getHostPort = (s: string): { host: string; port: number } => {
+    const lastColon = s.lastIndexOf(':');
+    const host = s.slice(0, lastColon);
+    const port = Number.parseInt(s.slice(lastColon + 1));
+    return { host, port };
+  };
+
+  const attachCommandFields = (
+    l: any,
+    ev: CommandStartedEvent | CommandSucceededEvent | CommandFailedEvent
+  ) => {
+    l.commandName = ev.commandName;
+    l.requestId = ev.requestId;
+    l.driverConnectionId = ev?.connectionId;
+    const { host, port } = getHostPort(ev.address);
+    l.serverHost = host;
+    l.serverPort = port;
+    l.serviceId = ev?.serviceId;
+
+    return l;
+  };
+
+  const attachConnectionFields = (l: any, ev: ConnectionPoolMonitoringEvent) => {
+    const { host, port } = getHostPort(ev.address);
+    l.serverHost = host;
+    l.serverPort = port;
+
+    return l;
+  };
+
+  let ev;
+  switch (logObject.name) {
```

Comment:
Using this as a discriminant allows for arbitrary objects with a name field to match these branches and is likely not the direction we want to go with this.

Reply:
Disregard as the logger is internal and will not be directly exposed to users.